kolena/_experimental/dataset/_dataset.py

```diff
@@ -90,6 +90,9 @@ def _infer_datatype(df: pd.DataFrame) -> Union[pd.DataFrame, str]:
 
 
 def _to_serialized_dataframe(df: pd.DataFrame) -> pd.DataFrame:
+    # Pandas defaults to int column labels if none provided. Cast to sort with added str labels
+    df.rename(mapper=str, axis="columns", copy=False, inplace=True)
```

Comment:
`_to_serialized_dataframe` probably shouldn't mutate the input `df`.

Reply:
I agree. I'm going to checkout https://github.com/kolenaIO/kolena/pull/273  since it looks to address this issue and will likely close this PR